app/debug_overlay.conf

```diff
@@ -1 +1,11 @@
 CONFIG_DEBUG=y
+CONFIG_ASSERT=y
+
+# Following options can be enabled additionally,
+# but will incur a higher runtime cost, so are thus
+# disabled by default.
+#
+# CONFIG_STACK_SENTINEL=y
+# CONFIG_SYS_HEAP_VALIDATE=y
+# CONFIG_SPIN_VALIDATE=y
+# CONFIG_SPIN_LOCK_TIME_LIMIT=50000
```

Comment:
@kv2019i I think we need to have 4 PR's here, one for each option and see where we go bang...